aria_data_loaders/aria_data_utils/spot_aria_p1_node.py

```diff
@@ -0,0 +1,591 @@
+                    marker_quaternion_form_robot_list.append(
+                        R.from_matrix(hand_mn_body_T_marker.rotation()).as_quat()
+                    )
+
+            cv2.imshow("hand_image", img_rend_hand)
+            cv2.waitKey(1)
+
+        marker_position_from_dock_np = np.array(marker_position_from_dock_list)
+        avg_marker_position_from_dock = np.mean(marker_position_from_dock_np, axis=0)
+
+        marker_quaternion_from_dock_np = np.array(marker_quaternion_form_dock_list)
+        avg_marker_quaternion_from_dock = np.mean(
+            marker_quaternion_from_dock_np, axis=0
+        )
+
+        marker_position_from_robot_np = np.array(marker_position_from_robot_list)
+        avg_marker_position_from_robot = np.mean(marker_position_from_robot_np, axis=0)
+
+        marker_quaternion_from_robot_np = np.array(marker_quaternion_form_robot_list)
+        avg_marker_quaternion_from_robot = np.mean(
+            marker_quaternion_from_robot_np, axis=0
+        )
+
+        avg_spotWorld_T_marker = sp.SE3(
+            R.from_quat(avg_marker_quaternion_from_dock).as_matrix(),
+            avg_marker_position_from_dock,
+        )
+        avg_spot_T_marker = sp.SE3(
+            R.from_quat(avg_marker_quaternion_from_robot).as_matrix(),
+            avg_marker_position_from_robot,
+        )
+
+        return avg_spotWorld_T_marker, avg_spot_T_marker
+
+    def get_avg_spotWorld_T_marker_HEAD(
+        self,
+        use_vision_as_world: bool = True,
+        data_size_for_avg: int = 10,
+        filter_dist: float = FILTER_DIST,
+    ):
+        pass
+
+    def get_avg_spotWorld_T_marker(
+        self,
+        camera: str = "hand",
+        use_vision_as_world: bool = True,
+        data_size_for_avg: int = 10,
+        filter_dist: float = FILTER_DIST,
+    ):
+        pass
+
+
+class SpotAriaP1Node:
+    """
+    TODO: Add docstring
+    """
+
+    def __init__(self, spot: Spot, verbose=False):
+        # Get Spot Skill Manager
+        self.skill_manager = SpotSkillManager(
+            should_stand_on_start=False,
+            use_mobile_pick=True,
+        )  # @FIXME: NEED THIS HERE AS IT INITIALIZES A Ros-node which we need here after
+
+        spot_qr = SpotQRDetector(spot=spot)
+        (
+            avg_spotWorld_T_marker,
+            avg_spot_T_marker,
+        ) = spot_qr.get_avg_spotWorld_T_marker_HAND()
+
+        # Instantiate static transform broadcaster for publishing marker w.r.t spotWorld transforms
+        self.static_tf_broadcaster = StaticTransformBroadcaster()
+
+        # Instantiate publisher for publishing goto pose for pick and place
+        self._goto_pose_pub_for_pick = rospy.Publisher(
+            "/goto_pose_for_pick", PoseStamped, queue_size=10
+        )
+        self._goto_pose_pub_for_place = rospy.Publisher(
+            "/goto_pose_for_place", PoseStamped, queue_size=10
+        )
+        self.goto_nav_pose_for_pick = None
+        self.goto_nav_pose_for_place = None
+
```

Comment:
renaming as discussed previously, variables should not have verbs like "goto" in their names

Reply:
addressed